wandb/sdk/launch/agent/agent.py

```diff
@@ -589,6 +589,21 @@ def _check_run_finished(
         try:
             run = job_tracker.run
             status = run.get_status().state
+
+            # Ensure run was not stopped intentionally
+            if status == "preempted":
+                try:
+                    run_state = RunState(
+                        self._api.get_run_state(
+                            self._entity, job_tracker.project, job_tracker.run_id
+                        )
+                    )
+                    if not run_state.is_alive:
+                        status = "stopped"
+                except wandb.CommError:
+                    # Run doesn't exist yet
+                    status = "failed"
```

Comment:
eventually we are going to want to make this at least an enum sheesh, maybe a TODO or ticket? I hate these raw strings (and on 607)

Reply:
Agreed. It would be a big enough change that it would bloat this particular PR, but I might do it as a follow-up. I'll add a TODO.